packages/cli-repl/src/run.spec.ts

```diff
@@ -97,4 +97,52 @@ describe('CLI entry point', function () {
       'MongoshInvalidInputError: [COMMON-10001] Invalid URI: /'
     );
   });
+
+  context('prompts for password', function () {
+    it('requests password when user is not redirecting output', async function () {
+      const args = [...pathToRun, 'mongodb://amy@localhost:27017'];
+      const proc = childProcess.spawn(process.execPath, args, {
+        stdio: ['pipe'],
+        env: { ...process.env, TEST_USE_STDOUT_FOR_PASSWORD: '1' },
+      });
+      let stdout = '';
+      let promptedForPassword = false;
+      proc.stdout?.setEncoding('utf8').on('data', (chunk) => {
+        stdout += chunk;
+        if (stdout.includes('Enter password')) {
+          promptedForPassword = true;
+          proc.stdin?.write('\n');
+        }
+      });
+
+      const [code] = await once(proc, 'exit');
+      expect(code).to.equal(1);
+      expect(promptedForPassword).to.be.true;
+    });
+
+    it('requests password when user is redirecting output', async function () {
+      const args = [
+        ...pathToRun,
+        'mongodb://amy@localhost:27017',
+        '> /dev/null',
```

Comment:
This doesn't work on Windows, does it? I think you'd need to end up doing something like `stdio: ['pipe', 'ignore', pipe']` or maybe actually opening a file and passing that fd as the stdout entry?

Reply:
yep, it does not! will check it!

Reply:
cleaned up in [83511f0](https://github.com/mongodb-js/mongosh/pull/2116/commits/83511f0fc5e6faf0a8a76c8d2c86f00e5e271b7b)